### PR TITLE
Reuse imagestreams with new-app

### LINKED ISSUE #20107: new-app creates unnecessary imagestream
https://github.com/openshift/origin/pull/20052 appears to have broken new-app's behavior.

Running in an empty project:
```
oc new-app https://github.com/openshift/nodejs-ex
```

creates an imagestreamtag named "nodejs:10" in the current namespace.

The buildconfig it creates references the openshift/nodejs:10 IST as expected, but the currentnamespace/nodejs:10 IST is also created.

### PR DESCRIPTION
**Pull request by coreydaley**
Trello: https://trello.com/c/BOR33gza/1559-3-reuse-existing-imagestreams-with-new-app-appcreationtechdebt
Fixes #20107

### CODE REVIEW

**Comment by coreydaley** [comment]
[(3) Reuse existing imagestreams with new-app [app_creation][tech_debt]](https://trello.com/c/BOR33gza/1559-3-reuse-existing-imagestreams-with-new-app-appcreationtechdebt)

**Comment by coreydaley** [comment]
Reuse existing imagestreams
```
$ oc new-build -D $'FROM node:8\nRUN echo "Test"' --name=node8
--> Found Docker image c5e9a81 (10 days old) from Docker Hub for "node:8"

    * An image stream tag will be created as "node:8" that will track the source image
    * A Docker build using a predefined Dockerfile will be created
      * The resulting image will be pushed to image stream tag "node8:latest"
      * Every time "node:8" changes a new build will be triggered

I0716 17:46:53.557823   26807 pipeline.go:191] ***** &app.ImageRef{Reference:reference.DockerImageReference{Registry:"", Namespace:"", Name:"node", Tag:"8", ID:""}, ResolvedReference:(*reference.DockerImageReference)(nil), AsResolvedImage:true, AsImageStream:true, OutputImage:false, Insecure:false, HasEmptyDir:false, TagDirectly:true, Tag:"", InternalDefaultTag:"", Env:app.Environment(nil), ObjectName:"", ContainerFn:(func(*core.Container))(nil), Stream:(*image.ImageStream)(nil), Info:(*docker10.DockerImage)(0xc4200c7918)}
--> Creating resources with label build=node8 ...
    imagestream.image.openshift.io "node" created
    imagestream.image.openshift.io "node8" created
    buildconfig.build.openshift.io "node8" created
--> Success
cdaley@workstation ~/go/src/github.com/openshift/origin (trello_1559_reuse_existing_imagestreams_take_2) [developer@https://127.0.0.1:8443]
$ oc new-build -D $'FROM node:10\nRUN echo "Test"' --name=node10 
--> Found Docker image 932354a (10 days old) from Docker Hub for "node:10"

    * An image stream tag will be created as "node:10" that will track the source image
    * A Docker build using a predefined Dockerfile will be created
      * The resulting image will be pushed to image stream tag "node10:latest"
      * Every time "node:10" changes a new build will be triggered

I0716 17:47:06.271540   27026 pipeline.go:191] ***** &app.ImageRef{Reference:reference.DockerImageReference{Registry:"", Namespace:"", Name:"node", Tag:"10", ID:""}, ResolvedReference:(*reference.DockerImageReference)(nil), AsResolvedImage:true, AsImageStream:true, OutputImage:false, Insecure:false, HasEmptyDir:false, TagDirectly:true, Tag:"", InternalDefaultTag:"", Env:app.Environment(nil), ObjectName:"", ContainerFn:(func(*core.Container))(nil), Stream:(*image.ImageStream)(nil), Info:(*docker10.DockerImage)(0xc4200c7518)}
--> Creating resources with label build=node10 ...
    imagestreamtag.image.openshift.io "node:10" created
    imagestream.image.openshift.io "node10" created
    buildconfig.build.openshift.io "node10" created
--> Success
cdaley@workstation ~/go/src/github.com/openshift/origin (trello_1559_reuse_existing_imagestreams_take_2) [developer@https://127.0.0.1:8443]
$ oc get is
NAME      DOCKER REPO                        TAGS      UPDATED
node      172.30.1.1:5000/myproject/node     10,8      3 seconds ago
node10    172.30.1.1:5000/myproject/node10             
node8     172.30.1.1:5000/myproject/node8              

```
Don't create extra imagestreams in current namespace
```
$ oc new-app https://github.com/openshift/nodejs-ex
...creation output...
$ oc get is
NAME        DOCKER REPO                           TAGS      UPDATED
nodejs-ex   172.30.1.1:5000/myproject/nodejs-ex
```

**Comment by coreydaley** [comment]
@openshift/sig-developer-experience ptal

File: pkg/oc/lib/newapp/app/pipeline.go [hunk 1]
```diff
@@ -194,6 +194,14 @@ func (p *Pipeline) Objects(accept, objectAccept Acceptor) (Objects, error) {
 		}
 		if objectAccept.Accept(repo) {
 			objects = append(objects, repo)
+		} else if len(p.InputImage.Reference.Namespace) == 0 {
```
Comment (bparees) [inline pkg/oc/lib/newapp/app/pipeline.go]: this seems weird.  can you add a comment explaining it?

and does it breakdown in scenarios where the input and output imagestreamtags are in the same namespace?  (and the imagestream may or may not already exist)

**Comment by coreydaley** [comment]
@bparees ptal, feedback addressed

**Comment by coreydaley** [comment]
@bparees ptal, i removed some un-needed changes from  pkg/oc/lib/newapp/app/imageref.go

**Comment by bparees** [comment]
this is still trying to create resources it shouldn't (hence the failing test):

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/20332/test_pull_request_origin_cmd/15817/

```
Standard error from the command:
    error: imagestreamtags.image.openshift.io is forbidden: User "test-user" cannot create imagestreamtags.image.openshift.io in the namespace "centos": User "test-user" cannot create imagestreamtags.image.openshift.io in project "centos": RBAC: clusterrole.rbac.authorization.k8s.io "basic-user2" not found
```

**Comment by coreydaley** [comment]
@bparees ptal, all tests are passing

**Pull request merged**